src/transport/grpc/mod.rs

```diff
@@ -73,11 +79,28 @@ pub enum GrpcTransportError {
     MessageHubProtoConversion(#[from] MessageHubProtoError),
 }
 
+pub(crate) struct NodeIdInterceptor(Uuid);
+
+impl NodeIdInterceptor {
+    pub(crate) fn new(uuid: Uuid) -> Self {
+        Self(uuid)
+    }
+}
+
+impl Interceptor for NodeIdInterceptor {
+    fn call(&mut self, mut req: Request<()>) -> Result<Request<()>, Status> {
+        let val = MetadataValue::from_str(&self.0.to_string())
+            .map_err(|_| Status::new(Code::InvalidArgument, "invalid metadata value"))?;
+        req.metadata_mut().insert("node_id", val);
```

Comment:
I would move this `node_id` string to a constant in the impl block of `NodeIdInterceptor`